packages/grid/src/vaadin-grid.js

```diff
@@ -509,13 +509,36 @@ class Grid extends ElementMixin(
       reorderElements: true,
     });
 
-    new ResizeObserver(() =>
+    // Memorize last scroll position in Firefox
+    if (this._firefox) {
+      this.$.table.addEventListener('scroll', () => {
+        const isVisible = !isElementHidden(this);
+        if (isVisible && this.__previousVisible !== false) {
+          this.__memorizedScrollTop = this.$.table.scrollTop;
```

Comment:
Could use `this._scrollTop` for getting and setting the value.

Reply:
Done